packages/camera/lib/new/src/camera_controller.dart

```diff
@@ -75,14 +84,35 @@ class CameraController {
     throw UnimplementedError('$defaultTargetPlatform not supported');
   }
 
-  /// Begins the flow of data between the inputs and outputs connected the camera instance.
+  /// Initializes the camera on the device.
+  ///
+  /// You must call [dispose] when you are done using the camera, otherwise it
+  /// will remain locked and be unavailable to other applications.
+  ///
+  /// Only one instance of [CameraController] can be active at a time. If you
+  /// call [initialize] on a [CameraController] while another is active, the old
+  /// controller will be disposed before initializing the new controller.
+  Future<void> initialize() {
+    if (_instance == this) {
+      return Future<void>.value();
+    }
+
+    _instance?.dispose();
```

Comment:
Should this `await` the dispose?

Reply:
Yea it does. But, I want this method to also be usable synchronously, so I don't want to make it `async`. I added a `Completer` to the method.

Also, I added tests to make sure the first controller is disposed.